recipes/libtiff/all/CMakeLists.txt

```
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
```

As this recipe is injecting CMake `WINDOWS_EXPORT_ALL_SYMBOLS `, it will require 3.4 at least to compile
```suggestion
cmake_minimum_required(VERSION 3.4)
```

---

Correct.

Let's also:
- move `WINDOWS_EXPORT_ALL_SYMBOLS` in CMake wrapper to make it clear.
- refactor a little bit `_patch_sources` which seems to have weird logic for MinGW.

---

Well no, moving WINDOWS_EXPORT_ALL_SYMBOLS globally is bad, I'll revert to previous patch